src/NRules/NRules.Fluent/Dsl/RuleBuildContext.cs

```diff
@@ -0,0 +1,44 @@
+﻿using NRules.Fluent.Expressions;
+using NRules.RuleModel.Builders;
+
+namespace NRules.Fluent.Dsl;
+
+internal readonly struct RuleBuildContext
+{
+    private RuleBuildContext(RuleBuilder builder)
+    {
+        Builder = builder;
+    }
+
+    public static RuleBuildContext CreateNew(RuleMetadata metadata)
+    {
+        var context = new RuleBuildContext(new RuleBuilder());
+        context.ApplyMetadata(metadata);
+        return context;
+    }
+
+    public static RuleBuildContext Empty = new();
+
+    public SymbolStack SymbolStack { get; } = new();
```

Comment:
@gritcsenko wouldn't this instantiate an empty SymbolStack for Empty? I think the idea is that it's null for Empty

Reply:
Fixed